examples/multimodal/components/encode_worker_onnx.py

```diff
@@ -0,0 +1,229 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from typing import AsyncIterator
+import os
+
+import onnxruntime as ort
+import numpy as np
+from PIL import Image
+import requests
+from io import BytesIO
+from transformers import LlavaProcessor # Needed only during init for preprocessing params
+
+from dynamo.sdk import depends, dynamo_endpoint, service
+# Assuming protocol definitions are in utils relative to components
+from utils.protocol import EncodeRequest, EncodeResponse
+
+logger = logging.getLogger(__name__)
+
+# --- Configuration --- (Adjust paths as needed for deployment)
+ONNX_MODEL_DIR = os.path.join(os.path.dirname(__file__), "..", "llava_onnx_encoder") # Relative path
+VISION_TOWER_ONNX_PATH = os.path.join(ONNX_MODEL_DIR, "llava_vision_tower.onnx")
+PROJECTOR_ONNX_PATH = os.path.join(ONNX_MODEL_DIR, "llava_projector.onnx")
+
+# Path to original model files (needed *only* for init to get processor params)
+# In a real deployment, you might hardcode these params or load from a config file
+# instead of loading the original processor here.
+ORIGINAL_MODEL_PATH = "/tmp/llava-1.5-7b-hf" # Or path where original files are accessible
+
+# Use "cuda" if GPU is available and configured, otherwise "cpu"
+# DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
+DEVICE = "cuda" # Assume GPU is always present
```

Comment:
nit: I think we could have some error handling if no GPU is present, or just set to cpu as the comment suggested